sdk/servicebus/azure-servicebus/samples/async_samples/deadletter_queue_async.py

```diff
@@ -0,0 +1,116 @@
+QUEUE_NAME = os.environ["SERVICE_BUS_QUEUE_NAME"]
+
+async def send_messages(servicebus_client, num_messages):
+    sender = servicebus_client.get_queue_sender(queue_name=QUEUE_NAME)
+    async with sender:
+        msg = [
+            ServiceBusMessage(
+                "Message to be deadlettered: {}".format(i),
+                subject="{}".format("Bad" if i % 2 == 0 else "Good"),
+            )
+            for i in range(num_messages)
+        ]
+        await sender.send_messages(msg)
+        print("Messages sent")
+
+async def exceed_max_delivery(servicebus_client):
+    receiver = servicebus_client.get_queue_receiver(queue_name=QUEUE_NAME)
+    dlq_receiver = servicebus_client.get_queue_receiver(queue_name=QUEUE_NAME, 
+                                                        sub_queue=ServiceBusSubQueue.DEAD_LETTER,
+                                                        prefetch_count=10)
+    async with receiver:
+        peek_msgs = await receiver.peek_messages()
+        while len(peek_msgs) > 0:
+            received_msgs = await receiver.receive_messages()
+            for msg in received_msgs:
+                print("Picked up message: delivery_count {}".format(msg.delivery_count))
+                await receiver.abandon_message(msg)
+            peek_msgs = await receiver.peek_messages()
+
+    async with dlq_receiver:
+        received_msgs = await dlq_receiver.receive_messages(max_message_count=1)
+        for msg in received_msgs:
+            print("Deadletter message:")
+            print(msg)
+            await dlq_receiver.complete_message(msg)
+    
+async def receive_messages(servicebus_client):
+    receiver = servicebus_client.get_queue_receiver(queue_name=QUEUE_NAME)
+    async with receiver:
+        received_msgs = await receiver.receive_messages(max_message_count=10)
+        for msg in received_msgs:
+            if msg.subject and msg.subject == "Good":
+                await receiver.complete_message(msg)
+            else:
+                await receiver.dead_letter_message(
+                    msg,
+                    reason="ProcessingError",
+                    error_description="Don't know what to do with this message.",
+                )
+
+async def fix_deadletters(servicebus_client):
+    sender = servicebus_client.get_queue_sender(queue_name=QUEUE_NAME)
+    receiver = servicebus_client.get_queue_receiver(queue_name=QUEUE_NAME)
+    dlq_receiver = servicebus_client.get_queue_receiver(queue_name=QUEUE_NAME, 
+                                                        sub_queue=ServiceBusSubQueue.DEAD_LETTER,
+                                                        prefetch_count=10)
+    msgs_to_send = []
+    async with dlq_receiver:
+        received_dlq_msgs = await dlq_receiver.receive_messages(max_message_count=10)
+        for msg in received_dlq_msgs:
+            if msg.subject and msg.subject == "Bad":
+                msg_copy = ServiceBusMessage(next(msg.body), subject="Good")
```

Comment:
You could use str(msg.body) which returns the string representation of the body
```suggestion
                msg_copy = ServiceBusMessage(str(msg.body), subject="Good")
```

Reply:
`str(msg.body)` returns the string `b'<generator object DataBody.data at 0x0000020682C79040>'`, but str(msg) works

Reply:
yeah.. msg.body is a generator object, str(message) should be good